### DISCUSSION TITLE
Using withoutTimestamps prevents casted dates in model events

### DISCUSSION THREAD

**Discussion by jasonvarga** [laravel/framework#48028] [Q&A]
I'm running into an issue that I'm hoping is just an oversight on my part.

I have an `updated` model event where I need the `updated_at` value:

```php
public static function booted()
{
    static::updated(function ($model) {
        dump($model->updated_at);
    });
}
```

Then elsewhere, I want to save the model without updating the timestamp:

```php
$model->foo = 'bar';
$model->withoutTimestamps(fn () => $model->save());
```

The `foo` attribute gets updated to `bar`, and the `updated_at` value doesn't get modified. Great!

However in the model event, `$model->updated_at` is just a string like `2023-08-11 03:27:38`. It doesn't get cast to a Carbon instance.

If I just did `$model->save()`, then the `updated_at` in the model event would be Carbon instance, but of course the updated_at value would also get updated.

Here's a full repo if it helps: https://github.com/jasonvarga/laravel-without-timestamps-issue/commit/086e0d0cff423904fe8260e2433589f71fa17067

Is there a way to have the best of both worlds? Be able to update the model without the timestamp, and be able to get the proper attribute in the event?

Thanks!

**Comment by MirAlexSky** [comment]
Hi! I have the same issue. Has anyone published this discussion as an issue? Maybe you've found a solution? It seems more like a bug to me.

I fixed it by adding "created_at" and "updated_at" to the cast array, so that the fields are forced to get cast.
